pkg/factorymodule/factory.go

```diff
@@ -0,0 +1,130 @@
+package factorymodule
+
+import (
+	"fmt"
+
+	"github.com/filecoin-project/mir/pkg/events"
+	"github.com/filecoin-project/mir/pkg/logging"
+	"github.com/filecoin-project/mir/pkg/modules"
+	"github.com/filecoin-project/mir/pkg/pb/eventpb"
+	"github.com/filecoin-project/mir/pkg/pb/factorymodulepb"
+	t "github.com/filecoin-project/mir/pkg/types"
+)
+
+// TODO: Add support for active modules as well.
+
+type moduleGenerator func(id t.ModuleID, params *factorymodulepb.GeneratorParams) modules.PassiveModule
```

Comment:
Maybe make it exported since it's already in the public interface (via a parameter)?

Reply:
Fixed.